### ISSUE TITLE
Style/SafeNavigation gives bad suggestion in if/else block

### ISSUE THREAD

**Issue by lordofthelake** [bbatsov/rubocop#3551]
The SafeNavigation cop has a false negative in case of an if/else block

## Expected behavior

Given this block of code:
```ruby
  def for_each_affected_element(&block)
      if some_precomputed_array
        some_precomputed_array.each(&block)
      else
        SomeModel
          .some_complicated_query_logic
          .some_other_methods
          .find_each(&block)
      end
    end
```

I expect not to receive a warning.

## Actual behavior

`Style/SafeNavigation` suggests to use safe navigation instead of checking in an object exists before calling the method.

## RuboCop version

```
$ rubocop -V
0.43.0 (using Parser 2.3.1.4, running on ruby 2.3.1 x86_64-darwin15)
```

**Comment by newdark** [comment]
I have a example that rubocop thinks is a problem as well
```ruby
        if ruby_gemset
          ruby_gemset.update!(used_at: DateTime.now)
        else
          ruby_gemset = RubyGemset.create!(gemset: gemset, version: version)
        end
```

**Comment by lordofthelake** [comment]
I think that in general if there is an "else" condition (or if it's part of a ternary operator expression), the check shouldn't apply.

Also, the autocorrect attempts to do very weird things, producing broken ruby code

**Comment by rrosenblum** [comment]
This was fixed as a part of #3517.
